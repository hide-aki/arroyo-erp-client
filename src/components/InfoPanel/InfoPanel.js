import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {Col} from 'react-bootstrap';
import FA from 'react-fontawesome';

/**
 * @name InfoPanel
 * @description Render a Panel wrapper with custom styling
 *
 */
class InfoPanel extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      hidden: this.props.hidden
        ? this.props.hidden && !this.props.static
        : false,
    };
  }

  handleHidden = () => {
    if (!this.props.static) {
      this.setState({
        hidden: !this.state.hidden,
      });
    }
  };

  render() {
    const _props = {};
    if (this.props.xs) {
      _props.xs = this.props.xs;
    }
    if (this.props.sm) {
      _props.sm = this.props.sm;
    }
    if (this.props.xs) {
      _props.md = this.props.md;
    }
    if (this.props.lg) {
      _props.lg = this.props.lg;
    }
    const barstyle = {
      cursor: this.props.static ? 'default' : 'pointer',
    };
    return (
      <Col {..._props} className="arroyo_box">
        <div className="arroyo_box_content">
          <div
            onClick={this.handleHidden}
            style={barstyle}
            className={`arroyo_box_header ${
              this.props.static ? '' : 'arroyo_box_header_hover'
            }`}
          >
            {this.props.title}
            {!this.props.static && (
              <FA name={`angle-${this.state.hidden ? 'down' : 'up'}`} />
            )}
          </div>

          <div
            className="arroyo_box_body"
            style={{
              display: this.state.hidden ? 'none' : 'block',
              width: '100%',
              padding: '10px',
              minHeight: this.props.height
                ? `${this.props.height}px`
                : 'inherit',
            }}
          >
            {this.props.children}
          </div>
        </div>
      </Col>
    );
  }
}

InfoPanel.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string.isRequired,
  hidden: PropTypes.bool,
  static: PropTypes.bool,
  height: PropTypes.number,
  xs: PropTypes.number,
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number,
};

export default InfoPanel;
