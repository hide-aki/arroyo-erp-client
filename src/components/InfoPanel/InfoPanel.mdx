---
name: InfoPanel
menu: Components
route: /components/InfoPanel
---

import { Props, Playground } from 'docz';
import InfoPanel from './InfoPanel';

# InfoPanel

This component is meant to render a Custom Panel.

<Props of={InfoPanel} />

<Playground>
  <InfoPanel title='Pendientes de aplicar' static>
    <div>Content as children</div>
  </InfoPanel>
</Playground>
