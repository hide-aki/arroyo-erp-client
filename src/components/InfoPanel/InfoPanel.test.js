// Container.test.js

import React from 'react';
import Enzyme, {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import InfoPanel from './InfoPanel';

Enzyme.configure({adapter: new Adapter()});

describe('<InfoPanel />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <InfoPanel hidden fluid="true" fixed="true" title="some title">
        Test
      </InfoPanel>
    );
  });

  it('Renders properly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('It has fluid prop', () => {
    expect(wrapper.find({fluid: true}));
  });
  it('It has fixed prop', () => {
    expect(wrapper.find({fluid: true}));
  });
  it('It has title prop', () => {
    expect(wrapper.find({title: 'some title'}));
  });

  it('It renders some text passed by children', () => {
    expect(wrapper.find('Test'));
  });

  describe('<InfoPanel hidden />', () => {
    let infoHidden;
    beforeAll(() => {
      infoHidden = mount(
        <InfoPanel hidden fluid="true" fixed="true" title="some title">
          Test
        </InfoPanel>
      );
    });

    it('Renders properly', () => {
      expect(toJSON(infoHidden)).toMatchSnapshot();
    });

    it('renders with hidden prop', () => {
      expect(infoHidden.find({hidden: false}));
    });
  });
});
