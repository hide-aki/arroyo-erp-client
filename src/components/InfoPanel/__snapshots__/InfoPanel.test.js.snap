// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoPanel /> <InfoPanel hidden /> Renders properly 1`] = `
<InfoPanel
  fixed="true"
  fluid="true"
  hidden={true}
  title="some title"
>
  <Col
    bsClass="col"
    className="nera_box"
    componentClass="div"
  >
    <div
      className="nera_box"
    >
      <div
        className="nera_box_content"
      >
        <div
          className="nera_box_header nera_box_header_hover"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          some title
          <FontAwesome
            name="angle-down"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-down"
            />
          </FontAwesome>
        </div>
        <div
          className="nera_box_body"
          style={
            Object {
              "display": "none",
              "minHeight": "inherit",
              "padding": "10px",
              "width": "100%",
            }
          }
        >
          Test
        </div>
      </div>
    </div>
  </Col>
</InfoPanel>
`;

exports[`<InfoPanel /> Renders properly 1`] = `
<InfoPanel
  fixed="true"
  fluid="true"
  hidden={true}
  title="some title"
>
  <Col
    bsClass="col"
    className="nera_box"
    componentClass="div"
  >
    <div
      className="nera_box"
    >
      <div
        className="nera_box_content"
      >
        <div
          className="nera_box_header nera_box_header_hover"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          some title
          <FontAwesome
            name="angle-down"
          >
            <span
              aria-hidden={true}
              className="fa fa-angle-down"
            />
          </FontAwesome>
        </div>
        <div
          className="nera_box_body"
          style={
            Object {
              "display": "none",
              "minHeight": "inherit",
              "padding": "10px",
              "width": "100%",
            }
          }
        >
          Test
        </div>
      </div>
    </div>
  </Col>
</InfoPanel>
`;
